Tests/ClientRuntimeTests/NetworkingTests/HashFunctionTests.swift

```diff
@@ -0,0 +1,150 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import XCTest
+@testable import ClientRuntime
+import AwsCommonRuntimeKit
+
+class HashFunctionTests: XCTestCase {
+
+    override func setUp() {
+        // Initialize function needs to be called before interacting with CRT
+        CommonRuntimeKit.initialize()
+    }
+
+    func testCRC32() {
+        guard let hashFunction = HashFunction.from(string: "crc32") else {
+            XCTFail("CRC32 not found")
+            return
+        }
+
+        // Create test data
+        let testBytes: [UInt8] = [0xFF, 0xFE, 0xFD, 0xFC] // Bytes not valid in UTF-8
+        let testData = Data(testBytes)
+
+        let result = try? hashFunction.computeHash(of: testData)?.toUInt32()
+        let expected = testData.computeCRC32()
```

Comment:
Would be better to test on hard coded values. You can use something like https://crccalc.com/?crc=0xFFFEFDFC&method=crc32&datatype=hex&outtype=0 to get the values.

Reply:
Changed as suggested